crates/sui-indexer/src/main.rs

```diff
@@ -38,55 +27,11 @@ async fn main() -> Result<(), IndexerError> {
         .unwrap(),
     );
 
-    let json_rpc_prometheus_registry = registry_service.default_registry();
-    let pg_connection_pool = new_pg_connection_pool(indexer_config.db_url.clone()).await?;
-    let handle = build_json_rpc_server(
-        &json_rpc_prometheus_registry,
-        pg_connection_pool.clone(),
-        &indexer_config.rpc_client_url,
-    )
-    .await
-    .expect("Json rpc server should not run into errors upon start.");
-    // let JSON RPC server run forever.
-    spawn_monitored_task!(handle.stopped());
-
-    retry(ExponentialBackoff::default(), || async {
-        let rpc_client = new_rpc_client(indexer_config.rpc_client_url.clone()).await?;
-        let pg_connection_pool = new_pg_connection_pool(indexer_config.db_url.clone()).await?;
-        // NOTE: Each handler is responsible for one type of data from nodes,like transactions and events;
-        // Handler orchestrator runs these handlers in parallel and manage them upon errors etc.
-        let handler_rpc_client = rpc_client.clone();
-        let handler_pg_pool = pg_connection_pool.clone();
+    let registry = registry_service.default_registry();
+    let pg_connection_pool = new_pg_connection_pool(&indexer_config.db_url).await?;
+    let store = PgIndexerStore::new(pg_connection_pool);
 
-        let prometheus_registry = registry_service.default_registry();
-        let handler_prometheus_registry = prometheus_registry.clone();
-        let handler_handle = spawn_monitored_task!(async move {
-            HandlerOrchestrator::new(
-                handler_rpc_client,
-                handler_pg_pool,
-                handler_prometheus_registry,
-            )
-            .run_forever()
-            .await;
-        });
-
-        let processor_prometheus_registry = prometheus_registry.clone();
-        let processor_handle = spawn_monitored_task!(async move {
-            ProcessorOrchestrator::new(
-                rpc_client.clone(),
-                pg_connection_pool,
-                processor_prometheus_registry,
-            )
-            .run_forever()
-            .await;
-        });
-
-        try_join_all(vec![handler_handle, processor_handle])
-            .await
-            .expect("Indexer main should not run into errors.");
-        Ok(())
-    })
-    .await
+    Indexer::start(&indexer_config.rpc_client_url, &registry, store).await
```

Comment:
`ProcessorOrchestrator` is for derivative tables, why remove it?

Reply:
all the derivative table processing should be done during index checkpoint
https://github.com/MystenLabs/sui/blob/987d73e6d0896710fbeb208123ddd609656bf90a/crates/sui-indexer/src/handlers/checkpoint_handler.rs#L164

it is more efficient since we have the data in memory already, having a separate process reading back data from DB/ from full node for extra processing seems inefficient.